substrate/frame/support/procedural/src/benchmark.rs

```diff
@@ -441,6 +444,31 @@ pub fn benchmarks(
 		benchmarks_by_name_mappings.push(quote!(#name_str => Self::#test_ident()))
 	}
 
+	let impl_test_function = if let Some(item) =
+		content.iter_mut().find(|item| matches!(item, Item::Macro(_)))
+	{
+		let Item::Macro(item_macro) = item else {
+			return Err(syn::Error::new(item.span(), "Expected an item macro, as found such item"));
+		};
+
+		if item_macro
+			.mac
+			.path
+			.segments
+			.iter()
+			.any(|s| s.ident == "impl_benchmark_test_suite")
+		{
+			let tokens = item_macro.mac.tokens.clone();
+			*item = Item::Verbatim(quote! {});
+
+			quote! { impl_test_function!((#( {} #benchmark_names )*)(#( #extra_benchmark_names )*)(#( #skip_meta_benchmark_names )*)#tokens); }
```

Comment:
Can you put a few new-lines in here please to make it easier to read?  
And why do we need the `{}`?

Reply:
Sure! I'll add those new lines.

The `quote! {}` in line [462](https://github.com/pandres95/polkadot-sdk/blob/bab21d25ab922f29114d3da956d88064edc3249c/substrate/frame/support/procedural/src/benchmark.rs#L462) is to suppress `impl_benchmark_test_suite` which otherwise left would generate the now existing `test_benchmarks` test function as well as the test for each benchmark function.

The `{}` behind `#benchmark_names` is necessary to match the input for the `impl_bench_case_tests` macro.